desc/integrals/singularities.py

```diff
@@ -327,104 +427,94 @@ def __call__(self, f, i):
             Data at source grid points to interpolate.
         i : int
             Index of polar node.
+        is_fourier : bool
+            Whether ``f`` holds Fourier coefficients as returned by
+            ``self.fourier``. Default is false.
+        vander : jnp.ndarray
+            Cached value for ``self.vander_polar(i)``.
 
         Returns
         -------
         fi : ndarray
             Source data interpolated to ith polar node.
-        """
-        return self._mat[:, i] @ f
```

Comment:
JAX would not perform this matrix multiplication correctly.

Reply:
I know that sounds suspicious, since it is a fundamental operation but that is what is observed. At first I presumed the pseudoinverse had a numerical issue, but even after replacing the `Ainv` matrix with a fourier transform the issue is persisted. Only once the `B` matrix was computed in chunks (self.polar_vander(i)) and multiplied against the Fourier coefficients was the issue resolved.

Since the $B$ matrix had to be computed in chunks to resolve the issue, it didn't make sense anymore to precompute the $A_{inv}$ matrix because evaluating the DFT interpolator then requires two matrix multiplies and then  $B[i] @ A_{inv} @ f$ which is basically the same cost as just $B[i] @ fourier_transform(f)$.

Reply:
if this is reproducible on a smaller MWE, it would be a good thing to make an issue for in the JAX repo